dotnet/samples/HuggingFaceImageTextExample/FormMain.cs

```diff
@@ -0,0 +1,205 @@
+
+        this._imageToTextService = this._kernel.GetRequiredService<IImageToTextService>();
+    }
+
+    /// <summary>
+    /// Main form load event.
+    /// </summary>
+    /// <param name="sender">The form main.</param>
+    /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
+    private void FormMain_Load(object sender, EventArgs e)
+    {
+        this.ChangeFolder();
+        this.Focus();
+    }
+
+    /// <summary>
+    /// Changes the folder and refreshes the images.
+    /// </summary>
+    private void ChangeFolder()
+    {
+        if (this.folderBrowserDialog1.ShowDialog() == DialogResult.OK)
+        {
+            this.lblImagesFolder.Text = $"Images folder: {this.folderBrowserDialog1.SelectedPath}";
+        }
+
+        if (string.IsNullOrEmpty(this.folderBrowserDialog1.SelectedPath))
+        {
+            MessageBox.Show("A folder needs to be selected.");
+
+            this.ChangeFolder();
+        }
+        else
+        {
+            this.RefreshImages();
+        }
+    }
+
+    /// <summary>
+    /// Refreshes the images in the flow layout panel.
+    /// </summary>
+    private void RefreshImages()
+    {
+        var imageDirectory = this.folderBrowserDialog1.SelectedPath;
+
+        var extensions = new List<string> { "*.jpg", "*.jpeg", "*.png", "*.gif", "*.bmp", "*.tiff", "*.ico", "*.svg" };
+        var myImagePaths = new List<string>();
+        foreach (var extension in extensions)
+        {
+            myImagePaths.AddRange(Directory.GetFiles(imageDirectory, extension, SearchOption.AllDirectories));
+        }
+
+        this.flowLayoutPanel1.Controls.Clear();
+        foreach (var imagePath in myImagePaths)
+        {
+            PictureBox pictureBox = new();
+            using var fs = new FileStream(imagePath, FileMode.Open, FileAccess.Read);
+            pictureBox.Image = new Bitmap(Image.FromStream(fs));
+            pictureBox.SizeMode = PictureBoxSizeMode.Zoom;
+            pictureBox.Height = 300;
+            pictureBox.Width = 300;
+            pictureBox.Click += this.PictureBoxOnClickAsync;
+            pictureBox.Tag = imagePath;
+            this.flowLayoutPanel1.Controls.Add(pictureBox);
+        }
+    }
+
+    /// <summary>
+    ///  Handles the Click event of the PictureBox control.
+    /// </summary>
+    /// <param name="sender">The picture box.</param>
+    /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
+#pragma warning disable VSTHRD100 // Avoid async void methods
+    private async void PictureBoxOnClickAsync(object? sender, EventArgs e)
+    {
+        this.textBox1.Text = "Processing...";
+        var pictureBox = ((PictureBox)sender!);
+        ImageContent imageContent = CreateImageContentFromPictureBox(pictureBox);
+        string text;
+        try
+        {
+            text = (await this._imageToTextService.GetTextContentAsync(imageContent).ConfigureAwait(false)).Text!;
```

Comment:
I think ConfigureAwait can be true in this case, because you want to UpdateImageDescription run on UI thread.